### DISCUSSION TITLE
How do you rewrite a component?

### DISCUSSION THREAD

**Discussion by filiphric** [slidevjs/slidev#1607] [Q&A] [answered]
I wasn’t able to find this in the docs, although I believe it’s mentioned somewhere that you can add **and rewrite** components.

Right now, when I add `components/H1.vue`, I’m getting an error: `[unplugin-vue-components] component "H1"(/components/h1.vue) has naming conflicts with other components, ignored.`

I’m trying to create custom theme.

**Comment by kermanx** [comment]
What if changing the name of the component to `Heading1`? `H1` may conflict with `<h1>` in HTML.

Also, you can do this in your slide to import a component explicitly, in case auto-import doesn't work:

```md
<script setup>
import YourComponent from './components/H1.vue'
</script>

<YourComponent />
```

**Reply by filiphric** [reply]
thanks for the suggestion 🙏  

My plan was to wrap the default `h1` component to create something like (just an example):

```
  <RoughNotation type="underline"><slot /></RoughNotation>
```

I’m trying to creat a theme, so I’d like to overwrite the `h1` instead of importing a separate component. So that when the theme is being used, the default `h1` becomes the custom `h1` provided by the theme

**Reply by kermanx** [reply, marked as answer]
I don't think overriding `h1` is possible... `h1` is an intrinsic element rather than a component. You can wrap `<h1>` manually in every slide or use CSS only instead. Also you can use [pre-parser](https://sli.dev/custom/config-parser) to transform

```md
# ABCD
````

into the following code:

```md
<MyH1>

# ABCD

</MyH1>

...
```

**Reply by filiphric** [reply]
Oh ok, thanks for the info. I thought it might work similarly to [Prose components](https://content.nuxt.com/components/prose#proseh1)
